## Commercial App Analysis For GooglePlay Store and App Store.

**A company is looking for the most downloadble Apps in both App store and IOS store before launching a project which aims to develop an application and want to know which genres attract users to download their new developed app.**
**The company is looking forward to develop a free English application**

### 1) Exploring Data

In [1]:
import pandas as pd
df_ios=pd.read_csv('AppleStore(1).csv') #Installing the IOSDataSet
df_android=pd.read_csv('googleplaystore.csv') #Installing theGooglePlayStoreDataSet

In [2]:
df_ios.head()

id               track_name  size_bytes currency  price  \
0  284882215                 Facebook   389879808      USD    0.0   
1  389801252                Instagram   113954816      USD    0.0   
2  529479190           Clash of Clans   116476928      USD    0.0   
3  420009108               Temple Run    65921024      USD    0.0   
4  284035177  Pandora - Music & Radio   130242560      USD    0.0   

   rating_count_tot  rating_count_ver  user_rating  user_rating_ver      ver  \
0           2974676               212          3.5              3.5     95.0   
1           2161558              1289          4.5              4.0    10.23   
2           2130805               579          4.5              4.5  9.24.12   
3           1724546              3842          4.5              4.0    1.6.2   
4           1126879              3594          4.0              4.5    8.4.1   

  cont_rating        prime_genre  sup_devices.num  ipadSc_urls.num  lang.num  \
0          4+  Social Networking               37                1        29   
1         12+      Photo & Video               37                0        29   
2          9+              Games               38                5        18   
3          9+              Games               40                5         1   
4         12+              Music               37                4         1   

   vpp_lic  
0        1  
1        1  
2        1  
3        1  
4        1

In [3]:
df_android.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [4]:
#Finding outmore about our GoogleStore data, understanding our datatypes.
df_android.info()  

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


We can find there are nulls in our dataset that we will be checking later in the next steps. Also Installs seems to have datatype of Object while it should be a float or integer number. We can change the datatype as well after fixing it.

In [5]:
#Finding outmore about our IOS data, understanding our datatypes.
df_ios.head()

id               track_name  size_bytes currency  price  \
0  284882215                 Facebook   389879808      USD    0.0   
1  389801252                Instagram   113954816      USD    0.0   
2  529479190           Clash of Clans   116476928      USD    0.0   
3  420009108               Temple Run    65921024      USD    0.0   
4  284035177  Pandora - Music & Radio   130242560      USD    0.0   

   rating_count_tot  rating_count_ver  user_rating  user_rating_ver      ver  \
0           2974676               212          3.5              3.5     95.0   
1           2161558              1289          4.5              4.0    10.23   
2           2130805               579          4.5              4.5  9.24.12   
3           1724546              3842          4.5              4.0    1.6.2   
4           1126879              3594          4.0              4.5    8.4.1   

  cont_rating        prime_genre  sup_devices.num  ipadSc_urls.num  lang.num  \
0          4+  Social Networking               37                1        29   
1         12+      Photo & Video               37                0        29   
2          9+              Games               38                5        18   
3          9+              Games               40                5         1   
4         12+              Music               37                4         1   

   vpp_lic  
0        1  
1        1  
2        1  
3        1  
4        1

In [6]:
#Checking nulls in our IOS data
df_ios.isnull().sum()

id                  0
track_name          0
size_bytes          0
currency            0
price               0
rating_count_tot    0
rating_count_ver    0
user_rating         0
user_rating_ver     0
ver                 0
cont_rating         0
prime_genre         0
sup_devices.num     0
ipadSc_urls.num     0
lang.num            0
vpp_lic             0
dtype: int64

We have no nulls values in our IOS Dataset.

In [7]:
#Checking nulls in our googlestoer dataset 
df_android.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

We can find that there are many null values in our dataset that we will need to check 

In [8]:
#Checking for duplicatesin our IOS dataset
df_ios.duplicated().sum()

0

We have no duplicates in our IOS DataSet

In [9]:
#Checking for duplicated in our android dataset
df_android.duplicated().sum()

483

We have many duplicates in our android dataset that we will need to discard

### 2) Cleaning Data

In [10]:
#We will have a closer look to the column we want to change itsdatatype
df_android['Installs'].value_counts()

1,000,000+        1579
10,000,000+       1252
100,000+          1169
10,000+           1054
1,000+             907
5,000,000+         752
100+               719
500,000+           539
50,000+            479
5,000+             477
100,000,000+       409
10+                386
500+               330
50,000,000+        289
50+                205
5+                  82
500,000,000+        72
1+                  67
1,000,000,000+      58
0+                  14
0                    1
Free                 1
Name: Installs, dtype: int64

We can note that there are many "," and "+" signs which notates that are over 5000 installations for example. We will predict that they are excatly 5000 installations so we will need to discard the "," and "+" signs also 'Free' is not relative to our column here so we will need to omit that as well. 

In [11]:
installs=list(df_android['Installs'])  #Creating a list with the number of installs 
new_installs=[] #Creating a new empty list to add our data with the correct syntax
for install in installs:
    new_install=install.replace(",","").replace("+","")   #omitting the "," and "+" signs
    new_installs.append(new_install) #returning the new list with the correct data we needed.

In [12]:
df_android['Installs']=new_installs  #Assigning our already exisiting 'Installs' column with the new correct data

#In the next step here, we will omitcheck which row exactly has "Free" in its number of Installs so we can delete it

df_android.query('Installs=="Free"') 

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

It's the row with index number of 10472 that we shall use to remove the unwanted data. 

In [13]:
#dropping data with row number 10472
df_android.drop(axis=0,index=10472,inplace=True)

In [14]:
#assigning a column with datatype float after fixing all issues.
df_android['Installs']=df_android['Installs'].astype(float)

In [15]:
#double-checking it was converted successfully. 
df_android['Installs'].describe()

count    1.084000e+04
mean     1.546434e+07
std      8.502936e+07
min      0.000000e+00
25%      1.000000e+03
50%      1.000000e+05
75%      5.000000e+06
max      1.000000e+09
Name: Installs, dtype: float64

We can note here that it shows exponential values in our data becausethe numbers are too large. That will be fixed later on when we analyze our data.

In [16]:
#dropping duplicates
df_android.drop_duplicates(inplace=True)

In [17]:
#doublechecking they were dropped successfully
df_android.duplicated().sum()

0

In [18]:
#In the next steps we will start replacing null values in out dataset
#We will replace them by calculating the mean(average) of all values in case of numbers
#In case of strings we will be getting the most common entry(mode)

rating_mean=df_android['Rating'].mean()

In [19]:
#Note that we are assigning a list here as the mode method also shows the datatype which we don't require here

type_mode=list(df_android['Type'].mode())
type_mode

['Free']

In [20]:
contentrating_mode=list(df_android['Content Rating'].mode())
contentrating_mode

['Everyone']

In [21]:
currentver_mode=list(df_android['Current Ver'].mode())
currentver_mode

['Varies with device']

In [22]:
androidver_mode=list(df_android['Android Ver'].mode())
androidver_mode

['4.1 and up']

In [23]:
#Filling all nulls with the values we just calculted
df_android['Rating'].fillna(value=rating_mean,inplace=True)
df_android['Type'].fillna(value=type_mode[0],inplace=True)
df_android['Content Rating'].fillna(value=contentrating_mode[0],inplace=True)
df_android['Current Ver'].fillna(value=contentrating_mode[0],inplace=True)
df_android['Android Ver'].fillna(value=androidver_mode[0],inplace=True)

In [24]:
#Doublechecking that there are no nulls
df_android.isnull().sum()

App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              0
Price             0
Content Rating    0
Genres            0
Last Updated      0
Current Ver       0
Android Ver       0
dtype: int64

### 3) Data Filtering

Our main goal here is to check free and english applications in both stores only. So we will need to filter both of our datasets using the above criteria. 

In [25]:
#Creating a new IOS dataset with free apps.
df_ios_free=df_ios.query('price==0.00')

In [26]:
#Creating a new googlestore dataset with free apps.
df_android_free=df_android.query('Type=="Free"')

In [27]:
#Creating a list with application names in both datasets
android_apps=list(df_android_free['App'])
ios_apps=list(df_ios_free['track_name'])

In [28]:
def isEnglish(string):
    """
    This function takes a string to check whether an app name
    is recoreded in English or not.
    """
    non_english = 0 
    for character in string:
        if ord(character) > 127:    #The ord() function returns the number representing the unicode code of a specified character.
            non_english += 1        #If the unicode is higher than 127 than it's not english 
    if non_english >=2:             #When detecting any foriegn character it should return False('not english')
        return False
    return True

In [29]:
#Creating a new list for Andriod english apps
android_eng_apps=[]
for androidapp in android_apps:  
    if isEnglish(androidapp) is True:   #If the application name is english, add it to the new list
        android_eng_apps.append(androidapp)
print(len(android_eng_apps))

9466


In [30]:
#Same goes to IOS apps
ios_eng_apps=[]
for iosapp in ios_apps:
    if isEnglish(iosapp) is True:
        ios_eng_apps.append(iosapp)
print(len(ios_eng_apps))

3169


In [31]:
#Creating a new dataframe with filtered data for android 
boolean_series1=df_android_free['App'].isin(android_eng_apps)
filtered_android_df=df_android_free[boolean_series1]

In [32]:
filtered_android_df

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                             Parkinson Exercices FR              MEDICAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

         Rating Reviews                Size    Installs  Type Price  \
0      4.100000     159                 19M     10000.0  Free     0   
1      3.900000     967                 14M    500000.0  Free     0   
2      4.700000   87510                8.7M   5000000.0  Free     0   
3      4.500000  215644                 25M  50000000.0  Free     0   
4      4.300000     967                2.8M    100000.0  Free     0   
...         ...     ...                 ...         ...   ...   ...   
10836  4.500000      38                 53M      5000.0  Free     0   
10837  5.000000       4                3.6M       100.0  Free     0   
10838  4.187877       3                9.5M      1000.0  Free     0   
10839  4.500000     114  Varies with device      1000.0  Free     0   
10840  4.500000  398307                 19M  10000000.0  Free     0   

      Content Rating                     Genres      Last Updated  \
0           Everyone               Art & Design   January 7, 2018   
1           Everyone  Art & Design;Pretend Play  January 15, 2018   
2           Everyone               Art & Design    August 1, 2018   
3               Teen               Art & Design      June 8, 2018   
4           Everyone    Art & Design;Creativity     June 20, 2018   
...              ...                        ...               ...   
10836       Everyone                  Education     July 25, 2017   
10837       Everyone                  Education      July 6, 2018   
10838       Everyone                    Medical  January 20, 2017   
10839     Mature 17+          Books & Reference  January 19, 2015   
10840       Everyone                  Lifestyle     July 25, 2018   

              Current Ver         Android Ver  
0                   1.0.0        4.0.3 and up  
1                   2.0.0        4.0.3 and up  
2                   1.2.4        4.0.3 and up  
3      Varies with device          4.2 and up  
4                     1.1          4.4 and up  
...                   ...                 ...  
10836                1.48          4.1 and up  
10837                 1.0          4.1 and up  
10838                 1.0          2.2 and up  
10839  Varies with device  Varies with device  
10840  Varies with device  Varies with device  

[9466 rows x 13 columns]

In [33]:
#Creating a new dataframe with filtered data for IOS 
boolean_series2=df_ios_free['track_name'].isin(ios_eng_apps)
filtered_ios_df=df_ios_free[boolean_series2]
filtered_ios_df

id                                    track_name  size_bytes  \
0      284882215                                      Facebook   389879808   
1      389801252                                     Instagram   113954816   
2      529479190                                Clash of Clans   116476928   
3      420009108                                    Temple Run    65921024   
4      284035177                       Pandora - Music & Radio   130242560   
...          ...                                           ...         ...   
7164  1065550288          Tchapper Messenger - Keep it private    82187264   
7177  1168960919                            Golden Lotus Slots   111412224   
7192  1170406182  Shark Boom - Challenge Friends with your Pet   245415936   
7194  1070052833                                    Go!Go!Cat!    91468800   
7195  1081295232   Suppin Detective: Expose their true visage!    83026944   

     currency  price  rating_count_tot  rating_count_ver  user_rating  \
0         USD    0.0           2974676               212          3.5   
1         USD    0.0           2161558              1289          4.5   
2         USD    0.0           2130805               579          4.5   
3         USD    0.0           1724546              3842          4.5   
4         USD    0.0           1126879              3594          4.0   
...       ...    ...               ...               ...          ...   
7164      USD    0.0                 0                 0          0.0   
7177      USD    0.0                 0                 0          0.0   
7192      USD    0.0                 0                 0          0.0   
7194      USD    0.0                 0                 0          0.0   
7195      USD    0.0                 0                 0          0.0   

      user_rating_ver      ver cont_rating        prime_genre  \
0                 3.5     95.0          4+  Social Networking   
1                 4.0    10.23         12+      Photo & Video   
2                 4.5  9.24.12          9+              Games   
3                 4.0    1.6.2          9+              Games   
4                 4.5    8.4.1         12+              Music   
...               ...      ...         ...                ...   
7164              0.0      2.5          4+          Lifestyle   
7177              0.0    1.1.0         17+              Games   
7192              0.0    1.0.9          4+              Games   
7194              0.0    1.1.2         12+              Games   
7195              0.0    1.0.3         12+      Entertainment   

      sup_devices.num  ipadSc_urls.num  lang.num  vpp_lic  
0                  37                1        29        1  
1                  37                0        29        1  
2                  38                5        18        1  
3                  40                5         1        1  
4                  37                4         1        1  
...               ...              ...       ...      ...  
7164               37                0         9        1  
7177               37                5         3        1  
7192               38                5         1        1  
7194               37                2         2        1  
7195               40                0         1        1  

[3169 rows x 16 columns]

### 4) Data Analysis: 

In [43]:
#getting the number of genres we have in the android dataset
len(list(filtered_android_df['Category'].value_counts().index))

33

In [46]:
#getting the number of genres we have in the IOS dataset
len(list(filtered_ios_df['prime_genre'].value_counts().index))

23

In [34]:
#Checking The Highest Genre in android Google play store.
filtered_android_df['Category'].value_counts()

FAMILY                 1730
GAME                   1022
TOOLS                   757
BUSINESS                414
PRODUCTIVITY            378
LIFESTYLE               347
FINANCE                 341
COMMUNICATION           337
MEDICAL                 322
SPORTS                  318
PERSONALIZATION         301
PHOTOGRAPHY             301
HEALTH_AND_FITNESS      290
SOCIAL                  277
NEWS_AND_MAGAZINES      256
TRAVEL_AND_LOCAL        223
SHOPPING                220
BOOKS_AND_REFERENCE     197
DATING                  186
VIDEO_PLAYERS           169
MAPS_AND_NAVIGATION     127
EDUCATION               125
FOOD_AND_DRINK          119
ENTERTAINMENT           108
LIBRARIES_AND_DEMO       83
AUTO_AND_VEHICLES        81
HOUSE_AND_HOME           77
WEATHER                  72
EVENTS                   63
ART_AND_DESIGN           61
PARENTING                56
COMICS                   55
BEAUTY                   53
Name: Category, dtype: int64

It shows that Family and Games are the highest genres in terms of number of apps in the Google Play Store

In [47]:
#Checking the average number of user ratings for each genre
round(filtered_android_df.groupby('Category')['Rating'].mean().nlargest(33),2)

Category
EVENTS                 4.36
EDUCATION              4.36
PARENTING              4.34
ART_AND_DESIGN         4.33
BOOKS_AND_REFERENCE    4.33
PERSONALIZATION        4.28
GAME                   4.27
BEAUTY                 4.26
SOCIAL                 4.25
HEALTH_AND_FITNESS     4.25
SHOPPING               4.24
WEATHER                4.23
SPORTS                 4.21
PRODUCTIVITY           4.20
PHOTOGRAPHY            4.19
FAMILY                 4.18
AUTO_AND_VEHICLES      4.18
HOUSE_AND_HOME         4.18
FOOD_AND_DRINK         4.18
LIBRARIES_AND_DEMO     4.18
MEDICAL                4.17
COMMUNICATION          4.16
COMICS                 4.15
FINANCE                4.14
BUSINESS               4.14
NEWS_AND_MAGAZINES     4.13
ENTERTAINMENT          4.12
TRAVEL_AND_LOCAL       4.10
LIFESTYLE              4.10
MAPS_AND_NAVIGATION    4.07
VIDEO_PLAYERS          4.07
TOOLS                  4.06
DATING                 4.02
Name: Rating, dtype: float64

Events, Education and Parenting have the highest ratings from users when it comes to android Google PlayStore

In [48]:
#Checking the average number of installs for each genre
round(filtered_android_df.groupby('Category')['Installs'].mean().nlargest(33),2)

Category
COMMUNICATION          71664410.06
SOCIAL                 45176396.76
VIDEO_PLAYERS          36809655.15
PRODUCTIVITY           32967273.32
PHOTOGRAPHY            32289929.62
GAME                   30635580.57
TRAVEL_AND_LOCAL       28527771.69
ENTERTAINMENT          22726481.48
NEWS_AND_MAGAZINES     21062875.82
TOOLS                  15126112.91
SHOPPING               11696948.11
BOOKS_AND_REFERENCE     9727836.35
PERSONALIZATION         6877818.56
WEATHER                 5898451.67
FAMILY                  5777851.55
MAPS_AND_NAVIGATION     5661104.57
SPORTS                  4802734.43
HEALTH_AND_FITNESS      4691544.49
EDUCATION               4258800.00
FINANCE                 2229188.07
FOOD_AND_DRINK          2149904.63
BUSINESS                2085512.29
ART_AND_DESIGN          2038050.82
HOUSE_AND_HOME          1618342.35
LIFESTYLE               1520110.75
DATING                  1104891.17
COMICS                   998475.45
LIBRARIES_AND_DEMO       757780.84
AUTO_AND_VE

Communication and social apps have the highest number of Installs when it comes to android Google Playstore

In [37]:
#Checking the highest number of genre in the Apple Store.
filtered_ios_df['prime_genre'].value_counts()

Games                1855
Entertainment         248
Photo & Video         160
Education             118
Social Networking     104
Shopping               80
Utilities              76
Sports                 69
Music                  65
Health & Fitness       63
Productivity           54
Lifestyle              49
News                   42
Travel                 36
Finance                35
Weather                27
Food & Drink           26
Business               17
Reference              17
Book                   12
Navigation              6
Medical                 6
Catalogs                4
Name: prime_genre, dtype: int64

Games seems to be ranking number one when it comes to number of application in the Apple Store

In [49]:
#Checking the avrage user ratings for each genre
round(filtered_ios_df.groupby('prime_genre')['rating_count_tot'].mean().nlargest(23),2)

prime_genre
Navigation           86090.33
Reference            79350.47
Social Networking    72916.55
Music                58205.03
Weather              54215.30
Book                 46384.92
Food & Drink         33333.92
Finance              32367.03
Travel               31358.50
Photo & Video        28441.54
Shopping             27816.20
Health & Fitness     24037.63
Sports               23008.90
Games                22985.21
Productivity         21799.15
News                 21750.07
Utilities            19900.47
Lifestyle            16739.35
Entertainment        14364.77
Business              7491.12
Education             7003.98
Catalogs              4004.00
Medical                612.00
Name: rating_count_tot, dtype: float64

Navigation and Reference have the highest count of  user ratings

In [50]:
round(filtered_ios_df.groupby('prime_genre')['user_rating'].mean().nlargest(23),2)

prime_genre
Catalogs             4.12
Productivity         4.06
Games                4.06
Shopping             3.98
Business             3.97
Music                3.95
Photo & Video        3.90
Health & Fitness     3.89
Reference            3.88
Navigation           3.83
Education            3.64
Food & Drink         3.63
Social Networking    3.60
Book                 3.58
Utilities            3.57
Entertainment        3.54
Weather              3.48
Finance              3.47
Lifestyle            3.46
Travel               3.44
News                 3.23
Sports               3.07
Medical              3.00
Name: user_rating, dtype: float64

Catalogs and Productivty and Games have the highest user ratings

### 5) Conclusion:

**GOOGLEPLAY:**
For Googe Play, the app genre I would recommend are communication apps as they have the highest number of installs. There are also 337 apps  in this category such that users might switch if a better app was produced as the average rating is 4.16

**IOS:**
For Apple Store, the app genre I would recommend are Game applications as they have highest count of apps. Also it has a very good user rating of average 4.06%. Comes in second place is social networking. The average rating count of users on social network application apps show that uers do show interest in using those kind of applications.

**Cross Platform:**
If we were looking to develop an app in both platforms so it's either Games or Communication/Social Networking that would win the bet.